src/main/java/graphql/execution/directives/QueryDirectivesImpl.java

```diff
@@ -69,7 +69,7 @@ private void computeValuesLazily() {
             BiMap<GraphQLDirective, Directive> directiveCounterParts = HashBiMap.create();
             BiMap<GraphQLDirective, QueryAppliedDirective> gqlDirectiveCounterParts = HashBiMap.create();
             BiMap<QueryAppliedDirective, GraphQLDirective> gqlDirectiveCounterPartsInverse = gqlDirectiveCounterParts.inverse();
-            mergedField.getFields().forEach(field -> {
+            mergedField.forEach(field -> {
```

Comment:
dont materlise the list view - forEach can handle single fields more efficiently without creating a list of 1